components/playlist/browser/playlist_service.cc

```diff
@@ -597,6 +597,37 @@ void PlaylistService::RemovePlaylist(const std::string& playlist_id) {
       {PlaylistChangeParams::Type::kListRemoved, playlist_id});
 }
 
+void PlaylistService::ResetAll() {
+  // Removes all saved data ----------------------------------------------------
+  // As these functions will be run in sync, it's safe to bind `this` to raw
+  // ptr.
+  // Remove all playlist items
+  GetAllPlaylistItems(base::BindOnce(
+      [](PlaylistService* service, std::vector<mojom::PlaylistItemPtr> items) {
+        for (const auto& item : items) {
+          for (auto playlist_id : item->parents) {
+            service->RemoveItemFromPlaylist(playlist_id, item->id);
+          }
+        }
+      },
+      this));
+
+  // Remove playlists except the default playlist
+  GetAllPlaylists(base::BindOnce(
+      [](PlaylistService* service, std::vector<mojom::PlaylistPtr> playlists) {
+        for (const auto& a_playlist : playlists) {
+          if (a_playlist->id != kDefaultPlaylistID) {
+            service->RemovePlaylist(*a_playlist->id);
+          }
+        }
+      },
+      this));
+
+  // Resets preference ---------------------------------------------------------
+  prefs_->ClearPref(kPlaylistCacheByDefault);
```

Comment:
I think clearing prefs first and then deleting items would be more safe.
If prefs are cleared but deleting files are failed, it'll be ok because we can recover it by handing orpahned items at startup?
but if it happens in opposite way, it could cause bugs.

Reply:
Or below works?
Clear prefs and then call `CleanUpOrphanedPlaylistItemDirs()`?

Reply:
Hum, I think removing the directory right away would be better.

Reply:
IMO, deleting metadat first and deleting file itself would be more safer because we can recover when metadata is deleted but file is not deleted. But can we recover in opposite situation (all files are deleted or partialy but metadata is live)?

Reply:
Yeah, I think we're on the same page. So what I wanted to say was 

1. Clear prefs
2. Remove everything in the playlist dir - we don't have to search for orphaned items.

Reply:
Yes, right. Even playlist dir is not deleted cleanly by any reasons, it would be recovered by `CleanUpOrphanedPlaylistItemDirs()`.

The reason what I mention about `CleanUpOrphanedPlaylistItemDirs()` above is I thought `CleanUpOrphanedPlaylistItemDirs()` will do deleting all files also because all files are already orphaned as prefs are cleared. I think nuke all files here also fine.

Reply:
I think deleting whole dir here would be better because `CleanUpOrphanedPlaylistItemDirs()` purpose is for recovering for unusual situation. but can you verify `CleanUpOrphanedPlaylistItemDirs()` works as expected when we only clear prefs here?

Reply:
> can you verify CleanUpOrphanedPlaylistItemDirs() works as expected when we only clear prefs here?

Sure, added unit test for the method.